### PR TITLE
Bubble sort Project

### PR DESCRIPTION
**Pull request by Zubenna**
# Summary
- Implemented method that takes array of numbers and sorts in ascending order
- Implemented another method that takes array of strings and sorts based on the length of string in ascending order

### CODE REVIEW

**Comment by Dandush03** [comment]
# Status: :red_circle: Required Change :red_circle: 
## Great work so far!

I'm getting an error while testing the app when the buble_sort_by received a block please try fixing this issue and submit again.

when running it like  `bubble_sort_by(%w[amazing, greeting,hello]) { |left, right| left.length - right.length } `i should recived an organize output like Hello amazing greeting

### Good Luck

### Happy Coding :computer: 

Daniel Laloush

**Comment by Zubenna** [comment]
Requested change has been implemented.

File: bubble_sort_by.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    n.times do |i|
+      result = yield arr[i], arr[i + 1]
+      if result >= 1
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        swapped = true
+      end
+    end
+    break if swapped == false
+  end
+end
```
Comment (shshamim63) [inline bubble_sort_by.rb]: bubble_sort_by should return a sorted array. But I can not see the arr declaration after line no. 13. So place 'arr' after line 13 and you are ready to go.

**Review by shshamim63 — APPROVED** [review]
## Status :1st_place_medal: Approved :tada: 
___
Hi @Zubenna and @Jmagero :wave: 
Almost covered the issue mentioned by the previous reviewer. You have to do a minor change on your bubble_sort_by. Check out the comment below in the block of code. As it is a minor issue, I am approving it in advance. Just make the change and you are ready to go.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/)

**Pull request merged**